src/pyscaffold/cli_parser.py

```diff
@@ -0,0 +1,182 @@
+        group = _ArgumentGroup(self, *args, **kwargs)
+        self._action_groups.append(group)
+        return group
+
+    def add_mutually_exclusive_group(self, **kwargs) -> _MutuallyExclusiveGroup:
+        group = _MutuallyExclusiveGroup(self, **kwargs)
+        self._mutually_exclusive_groups.append(group)
+        return group
+
+    def _get_prompt_actions(self, extensions: List[Extension]) -> List[Action]:
+        prompt_actions = []
+        for action in self.actions:
+            if action in self.noprompts:
+                continue
+            if action.dest == "help":
+                continue
+            if action in self.called_actions:
+                continue
+            # check if action was included by another, e.g. cirrus -> pre-commit
+            if is_included(action, extensions):
+                continue
+            prompt_actions.append(action)
+
+        return prompt_actions
+
+    def format_flag(self, action: Action) -> str:
+        flag = action.option_strings[-1]
+        formatter = self._get_formatter()
+        args = formatter._format_args(action, action.dest)
+        return f"{flag} {args}"
+
+    def prompt_user(self, opts: ScaffoldOpts) -> ScaffoldOpts:
+        """Prompt the user for additional options
+
+        ToDo: Give a nice error message if user cancels the prompting with ctrl+c
+        ToDo: Show the user the defaults from `actions.get_default_options` by
+         extracting this functionality into smaller functions that can be applied here
+        """
+        argument_msg = "Arguments of [{flag}] (enter for default)"
+        activate_msg = "Activate"
+        choices_msg = "Choices"
+
+        for action in self._get_prompt_actions(opts["extensions"]):
+            print(opts["extensions"])
+            if is_included(action, opts["extensions"]):
+                continue
+            flag = self.format_flag(action)
+            print("Flag:", flag)
+            print("Help:", action.help)
+
+            if action.nargs == 0:
+                input = inquirer.confirm(activate_msg, default=False)
+            elif action.nargs == "?":
+                input = inquirer.confirm(activate_msg, default=False)
+                if input:
+                    input = inquirer.text(
+                        argument_msg.format(flag=flag), default=action.default
+                    )
+                else:
+                    input = None
+            elif action.choices:
+                input = inquirer.list_input(choices_msg, choices=action.choices)
+            else:
+                input = inquirer.text(argument_msg.format(flag=flag), default=None)
+            opts = merge_user_input(opts, input, action)
+
+        return opts
+
+
+def is_included(action: Action, extensions: List[Extension]):
+    """Checks if action was included by another extension
+
+    An extension can include another using :obj:`~pyscaffold.extensions.include`
+
+    FIXME: This still doesn't work!
```

Comment:
Maybe by checking `action.options_string` and `ext.flag` instead of trying to get `action.const`? (the later is not guaranteed to exist)

Reply:
Right, I think this solution is way smarter :-)